File: surface/widgets/compose/components/widgets/form-controls/CheckboxGroup.kt
// ============================================================
// Clef Surface Compose Widget — CheckboxGroup
//
// Multi-choice selection from a visible list of checkboxes.
// Renders Material 3 Checkbox components in a Column or Row
// layout. Maps the checkbox-group.widget anatomy (root, label,
// items, item, itemControl, itemLabel) to Compose Checkbox
// with Row-based label alignment.
// ============================================================

package clef.surface.compose.components.widgets.formcontrols

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.selection.toggleable
import androidx.compose.material3.Checkbox
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.semantics.Role
import androidx.compose.ui.unit.dp

// --------------- Types ---------------

data class CheckboxGroupOption(
    val label: String,
    val value: String,
    val disabled: Boolean = false,
)

enum class CheckboxGroupOrientation { Horizontal, Vertical }

// --------------- Component ---------------

@Composable
fun CheckboxGroup(
    value: List<String>,
    options: List<CheckboxGroupOption>,
    onValueChange: (List<String>) -> Unit,
    modifier: Modifier = Modifier,
    label: String? = null,
    orientation: CheckboxGroupOrientation = CheckboxGroupOrientation.Vertical,
    enabled: Boolean = true,
) {
    Column(modifier = modifier) {
        if (label != null) {
            Text(
                text = label,
                style = MaterialTheme.typography.labelLarge,
            )
        }

        val content: @Composable () -> Unit = {
            options.forEach { option ->
                val isChecked = option.value in value
                val isEnabled = enabled && !option.disabled

                Row(
                    verticalAlignment = Alignment.CenterVertically,
                    modifier = Modifier.toggleable(
                        value = isChecked,
                        enabled = isEnabled,
                        role = Role.Checkbox,
                        onValueChange = { checked ->
                            val next = if (checked) {
                                value + option.value
                            } else {
                                value - option.value
                            }
                            onValueChange(next)
                        },
                    ),
                ) {
                    Checkbox(
                        checked = isChecked,
                        onCheckedChange = null, // handled by toggleable
                        enabled = isEnabled,
                    )
                    Spacer(modifier = Modifier.width(8.dp))
                    Text(
                        text = option.label,
                        style = MaterialTheme.typography.bodyMedium,
                        color = if (isEnabled) {
                            MaterialTheme.colorScheme.onSurface
                        } else {
                            MaterialTheme.colorScheme.onSurface.copy(alpha = 0.38f)
                        },
                    )
                }
            }
        }

        when (orientation) {
            CheckboxGroupOrientation.Vertical -> {
                Column { content() }
            }
            CheckboxGroupOrientation.Horizontal -> {
                Row(horizontalArrangement = Arrangement.spacedBy(16.dp)) { content() }
            }
        }
    }
}

File: surface/widgets/swiftui/components/widgets/form-controls/CheckboxGroup.swift
// ============================================================
// Clef Surface SwiftUI Widget — CheckboxGroup
//
// Multi-choice selection from a visible list of checkboxes.
// Renders Toggle or checkbox-style buttons in a vertical or
// horizontal layout. Maps the checkbox-group.widget anatomy
// (root, label, items, item, itemControl, itemLabel) to SwiftUI.
// ============================================================

import SwiftUI

// --------------- Types ---------------

struct CheckboxGroupOption: Identifiable {
    let id: String
    let label: String
    let value: String
    var disabled: Bool = false
}

enum CheckboxGroupOrientation { case horizontal, vertical }

// --------------- Component ---------------

/// CheckboxGroup view for multi-choice selection.
///
/// - Parameters:
///   - value: Binding to the set of selected values.
///   - options: Available checkbox options.
///   - label: Optional group label.
///   - orientation: Layout orientation.
///   - enabled: Whether the group is enabled.
struct CheckboxGroupView: View {
    @Binding var value: Set<String>
    var options: [CheckboxGroupOption]
    var label: String? = nil
    var orientation: CheckboxGroupOrientation = .vertical
    var enabled: Bool = true

    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            if let label = label {
                Text(label)
                    .font(.headline)
            }

            let content = ForEach(options) { option in
                let isChecked = value.contains(option.value)
                let isEnabled = enabled && !option.disabled

                SwiftUI.Button(action: {
                    guard isEnabled else { return }
                    if isChecked {
                        value.remove(option.value)
                    } else {
                        value.insert(option.value)
                    }
                }) {
                    HStack(spacing: 8) {
                        Image(systemName: isChecked ? "checkmark.square.fill" : "square")
                            .foregroundColor(isEnabled ? .accentColor : .gray)
                        Text(option.label)
                            .foregroundColor(isEnabled ? .primary : .gray)
                            .font(.body)
                    }
                }
                .buttonStyle(.plain)
                .disabled(!isEnabled)
            }

            if orientation == .horizontal {
                HStack(spacing: 16) { content }
            } else {
                VStack(alignment: .leading, spacing: 8) { content }
            }
        }
    }
}
